### ISSUE TITLE
2.0RC1

### ISSUE THREAD

**Issue by ChrisRM** [Appgutta/AGCalendar#40]
So, I've been working on this 2.0.RC1 release for a while, and wanted to post a status.
Im sure I've forgot alot of the changes Ive made, but this should be the most important fixes and additions. Im currently working on the last feature before I release it, which is the ability to select which calendar to add the event to (EventKit). 

That said, Im really considering submitting this module to the Appcelerator Marketplace, as I've laid down alot of work and time in it. What do you guys think ? Are there anyway to keep it open source, but still get the cash for some beers at the weekends ? ;-)
- Added `EKEventViewController` and `EKEventEditViewController` for the built-in event viewer/editor.  
- Added `showTable` to hide/show tableView under the calendar. (Default: true)  
- Added `showWeekNum` to hide/show week numbers on the first day of each week. (Default: false)  
- Added `nativeEditor` to use the built in `EKEventEditViewController` when clicking events in the table. (Default: true)  
- Added `theme` for easier skinning. (Default: Cal's theme) Fixes #26  
- Added `allDay` to set allDay events (Default: false) Fixes #21, #36  
- Added `event:deleted` which fires when an event is deleted using swipe in the tableview.  
- Fixed `fetchEvents` so it now responds with the list of events, including identifier. Fixes #38  
- Added `getCalendars()` to get a list of calendars (EventKit)  
- Added multiple dots to show how many events pr. date  
- Events now sorted by ascending order, by hour.  
- CalendarView tile-height set to 35, compared to 44 earlier. This gives better space below for the tableview.  
- Fixed #35 #issuecomment-14710880  
- Huge performance improvements

**Comment by jasonmorehouse** [comment]
This would be a welcome module in the marketplace, as the existing implementations are not great.  Of course, open source is nice too... but I'd be happy to contribute to your efforts.  If you don't end up adding it to the marketplace, put-up donate link so we can buy you some beers for the weekend!

-J

**Comment by aaronce** [comment]
sounds great so far chris!

**Comment by schmoove** [comment]
Yes- please put up a Paypal Donate button at least in the meantime- We're greatly appreciative of your efforts Chris!

**Comment by soyoh** [comment]
sound great, it would be awesome if is opensource, for sure i'll donate if you put some paypal donation buttons or something similar.
Thanks!

**Comment by ChrisRM** [comment]
I think I'll release version 2 in the marketplace and keep version 1 open source. I'll also merge the fixes from v2 to v1 so it functions correctly. Version 1 is more like the native calendar in iOS, so I think that would be the right thing to do.

Im also thinking about removing Core Data(sqlite really) from the available datasources in v2, as you can select which calendar to add events to in EventKit and also create new calendars, so theres really no need for an additional datasource.

Although v2 is heading in the marketplace direction, the ones interested here will get access to a private repository with the source.

**v2.0**

<img width="160" src="https://dl.dropbox.com/u/41389857/Photo%2026.03.13%2009%2000%2043.png" /> <img width="160" src="https://dl.dropbox.com/u/41389857/Photo%2026.03.13%2009%2000%2039.png" /> <img width="160" src="https://dl.dropbox.com/u/41389857/Photo%2026.03.13%2009%2000%2029.png" />

**Comment by clseva** [comment]
Hi Chris, I'm agree with you and I think your solution is the more balanced, specially if you merge new/pending features with actual version for now and keep this one open source in the future. Also, it is appreciated the private repository with new version for us or the "betatesters" :).

Thank you!

**Comment by Nubuu** [comment]
Hey Chris - looks awesome :) A private acces to give you feedback would be awesome!

One question:
Can we manipulate the data seen on the Edit screen? For example is it possible to insert own Repeat "types" or maybe just a extra field or something like that... ? :+1:

**Comment by ChrisRM** [comment]
The edit screen is not customizable, and I don't have any control over it. You have the option not to use it though. If you set `nativeEditor` to false, you can create your own custom view and insert the data provided by `event:clicked`.

Im in the work of setting up the private repository and inviting some betatesters. I want the module to be well tested before I upload it to the Marketplace. So... Those who want access and wish to contribute by testing version 2.0 of AGCalendar, please let me know. You'll get access to a private repository not only for this module, but some other modules in the work. Im about 90% done with a location based Augmented Reality module that also need some testing. Let me know

**Comment by nubu** [comment]
<< Would love to test it and give you feedback! ;-)

**Comment by clseva** [comment]
+1 :) Thank you!!

**Comment by aaronce** [comment]
Screenshots look awesome. I think you're right with keeping v1 and v2 separate. I already have a project lined up where v2 will be useful!

**Comment by jasonmorehouse** [comment]
Thanks Chris.  Please add me to the beta.  I'm happy to pay in the marketplace but would love to test the calendar in our own beta.

Cheers,
-Jason

**Comment by ChrisRM** [comment]
I've been busy at work the last week or so. I'll see if I can upload it within a couple of days. 

I'll send you some invitations as soon as the module is ready for some testing.

**Comment by clseva** [comment]
Hi Chris, any news?

**Comment by nubu** [comment]
Hey Chris, anything new for RC2.0 ? :) Would be soo awesome !

**Comment by ChrisRM** [comment]
Just wanted to say that I havent abonded the module, I just have so much to do at work these days. I'll try to make some time for it this weekend.

**Comment by Nubuu** [comment]
Hey Chris - any news about the module?

**Comment by ChrisRM** [comment]
Hi guys, finally I got some time to finish up this thing. I will be working on the module this week and through the weekend and push it early next week.

**Comment by ChrisRM** [comment]
Once again, sorry for the delay! I need testers for the new module, which is 100% customizable, includes a week and day view, week numbers and much more. All testers will get access to a private repository with all source files. Let me know

**Comment by schmoove** [comment]
Hi Chris,

Just thought I'd add a polite bump to this thread ;)
I'm sure you're plenty busy with other projects, but I'd be really interested to see how things are coming with the v2 module. I'd also like to volunteer some sponsorship for its development if that would help.. Cheers!
